<launch>
    <arg name = "veh" doc = "duckiebot"/>
    <arg name = "pkg_name" value = "motion_planner"/>
    <arg name = "node_name" default = "apriltag_detector_node"/>
    <arg name = "parameter_file_name" default = "default"/>
    <group ns = "$(arg veh)">
        <remap from = "apriltag_detector_node/initialise" to = "matlab/initialise"/>
        <remap from = "apriltag_detector_node/camera_info" to = "camera_node/camera_info"/>
        <remap from = "apriltag_detector_node/image/compressed" to = "camera_node/image/compressed"/>
        <node name = "$(arg node_name)" pkg = "$(arg pkg_name)" type = "$(arg node_name)" output = "screen">
            <param name = "veh" value = "$(arg veh)"/>
            <rosparam command = "load" file = "$(find motion_planner)/config/$(arg node_name)/$(arg parameter_file_name).yaml"/>
        </node>
    </group>
</launch>
